extra_data/reader.py

```diff
@@ -93,24 +95,27 @@ def __init__(self, filename, _cache_info=None):
             # This is used by the run files map.
             self.metadata_fstat = os.stat(self.file.id.get_vfd_handle())
 
-            # Close the file again - crude way to avoid hitting the limit of
-            # open files, which is only 1024 by default.
-            self.close()
-
         # {(file, source, group): (firsts, counts)}
         self._index_cache = {}
         # {source: set(keys)}
         self._keys_cache = {}
 
+    def __del__(self):
+        if self._file:
+            self._fc.close(self.filename)
```

Comment:
Can this call `self.close()`?

Reply:
Yes. That is kind of my personal style, but I don't care.